### PR TITLE
Add more histogram and exponential histogram tests

### PR DESCRIPTION
**Pull request by mx-psi**
### What does this PR do?

The first few commits add some missing tests based on coverage:
   - Add min/max to tests about histograms
   - Add test with empty histogram
   - Add test with unsupported combination of options
   - Add test for Cols

The last two commits add a new `pkg/internal/sketchtest` module for defining common test helpers for tests in `pkg/quantile` and `pkg/otlp/metrics`. These helpers help define the [quantile function](https://en.wikipedia.org/wiki/Quantile_function) of different common distributions. In particular I added the (truncated) normal and exponential distributions to have more real-world looking tests (since I would expect both of these to be common in the wild). The naming of parameters is based on how they appear on Wikipedia.

I then use this new module to add more tests, notably one on `pkg/otlp/metrics` that tests the accuracy of the output `quantile.Sketch` from a given Exponential Histogram.

### Motivation

Improve our test coverage for histograms and make sure our mapping is sound.

### CODE REVIEW

File: pkg/quantile/ddsketch_test.go [hunk 1]
```diff
@@ -7,19 +7,20 @@ package quantile
...
 	"github.com/DataDog/sketches-go/ddsketch"
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
+
+	"github.com/DataDog/opentelemetry-mapping-go/pkg/internal/sketchtest"
 )
 
 const (
-	acceptableFloatError = 2e-12
+	acceptableFloatError = 2e-11
```
Comment (mx-psi) [inline pkg/quantile/ddsketch_test.go:20]: Had to bump this after adding exponential and normal tests. I think it's just a fact that the `math.Exp` and `math.Erf`/`math.InvErf` functions do not have a lot of accuracy

File: pkg/otlp/metrics/sketches_test.go [hunk 2]
```diff
@@ -187,21 +195,27 @@ func TestHistogramSketches(t *testing.T) {
 	}
 }
 
-func TestExactSumCount(t *testing.T) {
+func TestExactSummaryStatistics(t *testing.T) {
```
Comment (songy23) [inline pkg/otlp/metrics/sketches_test.go:203]: Test name is a bit confusing as this is testing histogram rather than summary. Consider sth. like TestExactHistogramStats

File: pkg/internal/sketchtest/quantile.go [hunk 3]
```diff
@@ -0,0 +1,112 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2022-present Datadog, Inc.
+
+// package distrotest is an internal module with test helpers for generating points from a given distribution.
```
Comment (songy23) [inline pkg/internal/sketchtest/quantile.go:6]: distrotest -> sketchtest

**Review by songy23 — APPROVED** [review]
LGTM with some non-blocking comments

File: pkg/otlp/metrics/sketches_test.go (hunk 2 above)
Comment (mx-psi) [reply pkg/otlp/metrics/sketches_test.go]: Good point, I was using the term in [this sense](https://en.wikipedia.org/wiki/Summary_statistics) but it's definitely confusing. Renamed with your suggestion

File: pkg/otlp/metrics/sketches_test.go [hunk 4]
```diff
@@ -364,6 +364,17 @@ func TestInfiniteBounds(t *testing.T) {
 		name    string
 		getHist func() pmetric.Metrics
 	}{
+		{
+			name: "(-inf, inf): 0",
```
[the remark this answers is not in the document]
Comment (mx-psi) [reply pkg/otlp/metrics/sketches_test.go:374]: I added an `isEmpty` parameter to the tests to ensure that we do not accidentally skip tests because of malformed data
